### DISCUSSION TITLE
Increase route stops limit (currently 3)

### DISCUSSION THREAD

**Discussion by SievaFliesAlone** [organicmaps/organicmaps#482] [Ideas]
When planning a route, you're able to add stops to it. Currently you can add up to 3 of them (A, B, C).

It would be good if we could have **more**; it would be even better if there was **no limit**.

![Screenshot_20210603-114441_Organic_Maps](https://user-images.githubusercontent.com/790717/120696235-2bc2f180-c461-11eb-8c01-e3fd7ab53bb8.png)

**Comment by abtech42** [comment]
i'm agree with you.

**Comment by biodranik** [comment]
We have increased the stops limit in the last update to 100. Hope it's enough.

**Comment by 919nick** [comment]
Are there any plans to be able to change the order of the waypoints?

**Reply by biodranik** [reply]
What is your use-case? When and why do you need to change that?
P.S. iOS already supports rearranging.

**Reply by SievaFliesAlone** [reply]
When planning a route, it's useful to be able to rearrange waypoints. OM tries to build an optimal route (i.e. A > B > C) which sometimes is not what a user wants (i.e. A > C > B).

Let's say, I want to go to the **gas station** first, then I want to visit the **art gallery**, and then go to the **park**, but there's no way to change the order.
![image](https://user-images.githubusercontent.com/790717/147157363-670473b7-71ef-4dc0-a5de-4526232b8417.png)

**Reply by 919nick** [reply]
I'm testing the app for motorcycle navigation. I would be using the waypoints like shaping points to create a route. These routes are often circular for single day rides. When it tries to optimize the waypoints for me, I'm not able to maintain control of the route.

**Reply by biodranik** [reply]
@919nick can you please provide a route example (from another app), so we can understand why and what is not working in OM?

**Reply by 919nick** [reply]
Here's a link to a route that I was trying to duplicate in OM.
https://drive.google.com/file/d/12VMCFRIDH3rAiGmhhPbqpa3da9uv3EpP/view?usp=sharing

After a few attempts I did get OM routing to follow this route.  It's just that if it reorders the waypoints I have to delete them and try again.  It would be much easier to be able to control the order of the waypoints.

**Reply by biodranik** [reply]
Thanks, clear! What would be the best UX way to add stops manually, and to help build the most optimal path?

**Reply by 919nick** [reply]
I think the Kurviger app does a good job of this.  You can manually select a waypoint or shaping point and drag it around, or display a list of the waypoints and reorder them.  In my opinion, what I like so much about OM is the clean and simple interface.  Just being able to manually drag a waypoint to a spot would be enough.  Off course saving the route once it's created would be even better!  :)

**Reply by pastk** [reply]
A classic approach to it is to make the points re-arrangeable by user, but it will require the most development effort I guess.

The easiest way to do it is just do not rearrange the stops automatically - leave them in the order user has added them. That's the default behavior of the most navigation apps.

However some users might want the automatic re-arrangement feature still, e.g. when adding a lot of intermediate points in random order to make sure the route adheres to some wanted shape. Maybe add a button like "optimize stops order" or a checkbox "ignore stops order"...
